packages/core/src/types.ts

```diff
@@ -40,13 +40,21 @@ export interface IValidator<Resource, Input, Config, Output> {
   validateOutput?: (opts: { resource: Resource; output?: Output; config?: Config }) => IPrismDiagnostic[];
 }
 
-export interface IPrismComponents<Resource, Input, Output, Config extends IPrismConfig> {
+type MockerOrForwarder<Resource, Input, Output, Config extends IPrismConfig> =
+  | {
+      forwarder?: IForwarder<Resource, Input, Config, Output>;
```

Comment:
```suggestion
```

If it's mocker OR forwarder then why still make forwarder an option?

Reply:
That's the way you do this in TypeScript. You need to make two types and union them where each of them is 

See an example:

https://www.typescriptlang.org/play/index.html#code/C4TwDgpgBAhg+gewE5wEZQLxQN6wPwBcUAzsEgJYB2A5gNxSpGUCuAtqhElAL5QA+OWEVIUa9VISgt2nHrQBQ8gMYJKpKAgCMReMjSYc8qEKgAiABYQQp+dwXLV6hACYdiFOizYjDIpoAMtvYA9MFQAKJISMgANFKq0OTEUGDRAG7kACYQmQ5qwBoAzG56njh28kA

However your comment made me realise there's probably a more clear way that I pushed in 5c3edf9